pulsar-broker/src/test/java/org/apache/pulsar/client/api/ConsumerRedeliveryTest.java

```diff
@@ -424,4 +424,24 @@ public void testAckNotSent(int numAcked, int batchSize, CommandAck.AckType ackTy
             assertTrue(values.isEmpty());
         }
     }
+
+    @Test
+    public void testRedeliverMessagesWithoutValue() throws Exception {
+        String topic = "persistent://my-property/my-ns/testRedeliverMessagesWithoutValue";
+        @Cleanup Consumer<Integer> consumer = pulsarClient.newConsumer(Schema.INT32)
+                .topic(topic)
+                .subscriptionName("sub")
+                .enableRetry(true)
+                .subscribe();
+        @Cleanup Producer<Integer> producer = pulsarClient.newProducer(Schema.INT32)
+                .topic(topic)
+                .enableBatching(true)
+                .create();
+        for (int i = 0; i < 10; i++) {
+            producer.newMessage().key("messages without value").send();
+        }
+
+        Message<Integer> message = consumer.receive();
+        consumer.reconsumeLater(message, 2, TimeUnit.SECONDS);
```

Comment:
It's better to add an `assert` to ensure the message could be redelivered later.